internal/command/deploy/machines.go

```diff
@@ -90,6 +91,9 @@ func NewMachineDeployment(ctx context.Context, args MachineDeploymentArgs) (Mach
 	if err != nil {
 		return nil, err
 	}
+	if args.NewVolumeName != "" {
+		appConfig.Mounts.Source = args.NewVolumeName
```

Comment:
Could `appConfig.Mounts` ever be nil?

Reply:
appConfig is fetched from the context, it is simpler to override this at the caller's appConfig instance instead of patching it here as a side effect.